src/effects/ColorDepthEffect.ts

```diff
@@ -0,0 +1,98 @@
+import { Uniform } from "three";
+import { Effect } from "./Effect.js";
+
+import fragmentShader from "./shaders/color-depth.frag";
+
+/**
+ * ColorDepthEffect options.
+ *
+ * @category Effects
+ */
+
+export interface ColorDepthEffectOptions {
+
+	/**
+	 * The color bit depth.
+	 *
+	 * @defaultValue 16
+	 */
+
+	bits?: number;
+
+}
+
+/**
+ * A color depth effect.
+ *
+ * Simulates a hardware limitation to create a retro feel. The real color depth remains unchanged.
+ *
+ * @category Effects
+ */
+
+export class ColorDepthEffect extends Effect {
+
+	/**
+	 * The current amount of bits.
+	 */
+
+	private _bits: number = 0;
+
+	/**
+	 * Constructs a new color depth effect.
+	 *
+	 * @param options - The options.
+	 */
+
+	constructor({ bits = 16 }: ColorDepthEffectOptions = {}) {
+
+		super("ColorDepthEffect");
+
+		this.fragmentShader = fragmentShader;
+
+		const uniforms = this.input.uniforms;
+		uniforms.set("factor", new Uniform(Math.pow(2.0, bits / 3.0)));
```

Comment:
```suggestion
		uniforms.set("factor", new Uniform(0.0));
```

Reply:
That makes much more sense as we set the value on the next line down anyway! Cheers, have amended the PR.